### PR TITLE
blueprint for table component

### PR DESCRIPTION
**Pull request by devendra940**
Blueprint for the table component, this will show us the basic structure for table component with the feature that we are including in the table component

### CODE REVIEW

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		sort:true,
+		fieldType:select,
+		fieldTypeOptions: [{key:"",value:""}]
+	}],
+	checkboxEnableColumn: ["col1"],
+	columnSelectionDropdown: {
+		defaultColumns: ["col1", "col2"]
+	},
+	data: data,
+	totalRecords:  totalRecords,
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Remove extra space before _totalRecords_. Looks like there are 2 spaces.

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
+TableData = {
+	columnHeaders :[{
+		fieldName: "col1",
+		fieldLabel: "col1",
+		index:0,
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Add space after :

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		sort:true,
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Add space after :

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+		fieldType: text,
+		},
+		{
+		fieldName: "col2",
+		fieldLabel: "col2",
+		index:1,
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Add space after :

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		sort:true,
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Add space after :

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+		fieldType:select,
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Add space after :

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+		fieldTypeOptions: [{key:"",value:""}]
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Add space after colons(:) and one after the comma(,). It should be something like this [{key: "", value: ""}]

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+	}],
+	checkboxEnableColumn: ["col1"],
+	columnSelectionDropdown: {
+		defaultColumns: ["col1", "col2"]
+	},
+	data: data,
+	totalRecords:  totalRecords,
+	sortOrder: {order:"ascending", column: "col1"},
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Add space after colons(:)

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+	backendOperations: true,
+	pagination: true,
+	linkEditOption : "col1",
+	newRowInsertion: [{
+		buttonLabel: "New Item",
+		action: "add"
+	}],
+	actionCol: [{
+	 buttonLabel : "Edit",
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Remove extra space before :

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+	 action: "edit"
+	},{
+	 buttonLabel : "Delete",
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Remove extra space before :

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+	 action: "delete"
+	}]
+	,defaultRows : 10
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Remove extra space before :

File: src/datatable-blueprint/tabledata-blueprint.json [hunk 12]
```diff
@@ -0,0 +1,39 @@
+TableData = {
+	columnHeaders :[{
```
Comment (manmohan-b) [inline src/datatable-blueprint/tabledata-blueprint.json]: Please correct overall formatting and the indentation. You can try some linters for example https://jsonlint.com/

**Comment by han-so** [comment]
@kaseyllabelle @manmohan-b @ameyk-cybage 
The JSON object with table data will be created wherever the table component is being used, in our case "app-component.js". So no use of creating a separate file and adding it to our project.
We will discuss further on the JSON object and raise another PR with all respective code changes.
Closing this PR. Thanks!

**Pull request closed without merging**